### PR TITLE
[DOC] API doc

### LINKED ISSUE #602: model zoo download not respecting MXNET_HOME
Currently, most of the datasets are respecting the MXNET_HOME variable for default path, but the models are not.

### PR DESCRIPTION
**Pull request by szha**
## Description ##
Add example code for data API

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] Add example code for data API
- [x] make PR linkcheck optional
- [x] make model API respect MXNET_HOME

## Comments ##
- #405 
- resolves #602

### CODE REVIEW

File: src/gluonnlp/data/word_embedding_evaluation.py [hunk 1]
```diff
@@ -709,6 +758,37 @@ class GoogleAnalogyTestSet(WordAnalogyEvaluationDataset):
 
     License: Unspecified
 
+    Each sample consists of two analogical pairs of words.
+
+    Parameters
```
Comment (szha) [inline src/gluonnlp/data/word_embedding_evaluation.py]: @leezu some parameters are missed from the analogy classes. was it intentional? I added it for only GoogleAnalogy but not yet for BiggerAnalogy.

File: src/gluonnlp/data/word_embedding_evaluation.py [hunk 2]
```diff
@@ -589,7 +627,7 @@ class BakerVerb143(WordSimilarityEvaluationDataset):
     _verify_ssl = False  # ie.technion.ac.il serves an invalid cert as of 2018-04-16
 
     min = 0
-    max = 10
+    max = 1
```
Comment (szha) [inline src/gluonnlp/data/word_embedding_evaluation.py]: @leezu would this affect the evaluation results?

**Comment by mli** [comment]
Job PR-614/16 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-614/16/index.html

File: src/gluonnlp/data/word_embedding_evaluation.py [hunk 3]
```diff
@@ -355,42 +392,23 @@ class SimLex999(WordSimilarityEvaluationDataset):
...
-      are taken from the University of South Florida Free Association Dataset.
-    - SimAssoc333: Binary indicator of whether the pair is one of the 333 most
-      associated in the dataset (according to Assoc(USF)). This subset of
-      SimLex999 is often the hardest for computational models to capture
-      because the noise from high association can confound the similarity
-      rating. See the paper for more details.
-    - SD(SimLex): The standard deviation of annotator scores when rating this
-      pair. Low values indicate good agreement between the 15+ annotators on
-      the similarity value SimLex999. Higher scores indicate less certainty.
+    Each sample consists of a pair of words, and a score with scale from
```
Comment (eric-haibin-lin) [inline src/gluonnlp/data/word_embedding_evaluation.py]: Why removing these?

Comment (szha) [reply src/gluonnlp/data/word_embedding_evaluation.py]: they are no longer accurate as the dataset only returns a subset of the fields with no way to return all of them.

Comment (szha) [reply src/gluonnlp/data/word_embedding_evaluation.py]: cc @leezu

File: src/gluonnlp/data/word_embedding_evaluation.py [hunk 4]
```diff
@@ -190,7 +202,7 @@ def _get_data(self):
                     self.root,
                     'wordsim353_sim_rel/wordsim_similarity_goldstandard.txt'))
 
-        return list({tuple(row) for row in CorpusDataset(paths)})
+        return sorted(list({tuple(row) for row in CorpusDataset(paths)}))
```
Comment (sxjscience) [inline src/gluonnlp/data/word_embedding_evaluation.py]: Why sorted?

Comment (szha) [reply src/gluonnlp/data/word_embedding_evaluation.py]: it's generating a list from a set, which doesn't guarantee order. Doing a sort here could help make the result consistent and the doctest easier.

Comment (sxjscience) [reply src/gluonnlp/data/word_embedding_evaluation.py]: I see.

File: src/gluonnlp/data/word_embedding_evaluation.py (hunk 2 above)
Comment (leezu) [reply src/gluonnlp/data/word_embedding_evaluation.py]: @szha, I'm not aware of any code relying on this attribute. So at least for the code in GluonNLP, correcting this value will not make any problems. Thanks for spotting it!

File: src/gluonnlp/data/word_embedding_evaluation.py (hunk 1 above)
Comment (leezu) [reply src/gluonnlp/data/word_embedding_evaluation.py]: The missing documentation was not intentional. Thanks for fixing!

**Comment by leezu** [comment]
Thanks @szha!

**Pull request merged**